File: coding-test-4.ts
const money_list_ts = [50000, 10000, 500, 100, 50, 10];
const money_list_kor_ts = [
  "오만원권",
  "만원권",
  "오백원권",
  "백원권",
  "오십원권",
  "십원권",
];
let change_list_ts: any = [];
function change_money_ts(pay_money, total_price) {
  let money = pay_money - total_price;
  //console.log(money);
  for (let i = 0; i < money_list_ts.length; i++) {
    if (money > money_list_ts[i]) {
      //console.log(`${money_list[i]}원 : ${parseInt(money / money_list[i])}개`);
      change_list_ts.push(`${Math.floor(money / money_list_ts[i])}`);
      money = money - money_list_ts[i] * Math.floor(money / money_list_ts[i]);
      //console.log(money);
    } else {
      //console.log(`${money_list[i]}권 : 0개`);
      change_list_ts.push(0);
    }
  }
  const result = money_list_kor_ts.reduce((acc, curr, idx) => {
    acc[curr] = change_list_ts[idx];
    return acc;
  }, new Object());
  return console.log(result);
}

change_money_ts(100000, 34830);

File: test/coding-test-4.py
# 지불금액: pay_money
# 총 합계: total_price
# 거스름돈: change_money

money_list = [50000, 10000, 500, 100, 50, 10]
change_list = []


def change_money(pay_money, total_price):
    money = pay_money - total_price
    i = 0

    while i < len(money_list):
        if (money > money_list[i]):
            change_list.append(money // money_list[i])
            money = money - money_list[i] * (money // money_list[i])
        else:
            change_list.append(0)
        i = i+1
    return print(change_list)


change_money(100000, 34830)
